Reconfigure inputs for actions (#37)
bufbuild/buf-action: README.md

[lint](https://buf.build/docs/lint/overview),
[format](https://buf.build/docs/format/style),
and [breaking change](https://buf.build/docs/breaking/overview) errors,

> as well as to automatically [publish schema changes](https://buf.build/docs/bsr/module/publish) to the [Buf Schema Registry (BSR)](https://buf.build/product/bsr).

Reviewer: IIRC we want first instance to be what you have here, and then all other instances of "Buf Schema Registry" to be just "BSR"

Author: Fixed.

          token: ${{ secrets.BUF_TOKEN }}
```

### Default behavior

> When you push a commit, tag, or branch to GitHub, the action will [push named modules to the BSR](https://buf.build/docs/bsr/module/publish).
> On a pull request, the action will run all checks and post a [summary comment](#summary-comment) on the PR.
> When you delete a branch or tag, the action will archive the corresponding label on the BSR.
> This behavior can be customized by setting the action inputs.

Reviewer: Be explicit when we are referring to Git primitives (vs BSR) and also IIRC @bufdev like explicitly referencing the `buf` commands being used.

```suggestion
When you push a Git commit, tag, or branch to GitHub, the action will [push named modules to the BSR](https://buf.build/docs/bsr/module/publish) using `buf push`.

On a pull request, the action will run all checks and post a [summary comment](#summary-comment) on the PR using `buf build`, `buf lint`, `buf format`, and `buf breaking`.

When you delete a Git branch or tag, the action will archive the corresponding label on the BSR.

This behavior can be configured to fit your desired workflow.
```

### Configuration

To customize the behavior of the action, you can set the following inputs in the workflow file.
Add these inputs under the `with` section of the `uses` step in the workflow file.

```yaml
- uses: bufbuild/buf-action@v0.1
  with:
    ...
```

| Parameter                       | Description                                        | Default            |
|:--------------------------------|:---------------------------------------------------|:-------------------|

> | `archive`                       | Whether to run the archive step. | Runs on Git deletes |

Reviewer: Looks like this is alphabetical order? IIRC we wanted to order this most important first.

Reviewer: @bufdev can you remind us what you wanted here?

Author: Changed ordering.

| `domain`                        | Domain for logging into the BSR, enterprise only.| `buf.build` |
| `exclude_imports`               | Exclude imports. | |
| `exclude_paths`                 | Exclude specific files or directories (separated by newlines). | |
| `format`                        | Whether to run the formatting step. | Runs on pushes to Git PR |
| `input`                         | [Input](https://buf.build/docs/reference/inputs) for the `buf` command. | |
| `lint`                          | Whether to run the linting step. | Runs on pushes to Git PR |
| `paths`                         | Limit to specific files or directories (separated by newlines). | |
| `pr_comment`                    | Comment the results on the pull request. | Only on pull requests |
| `push`                          | Whether to run the push step. | Runs on Git pushes |
| `push_disable_create`           | Disables repository creation if it does not exist. | False |
| `setup_only`                    | Setup only the `buf` environment, optionally logging into the BSR, but without executing other commands. | |

> | `token`                         | API token for logging into the BSR. | |

Reviewer: ```suggestion
| `token`                         | API token for [logging into the BSR](https://buf.build/docs/bsr/authentication). | |
```

### Configuration

To customize the behavior of the action, you can set the following parameters in the workflow file.
Add these parameters under the `with` section of the `uses` step in the workflow file.

```yaml
- uses: bufbuild/buf-action@v0.1
  with:
    ...
```

| Parameter                       | Description                                        | Default            |
|:--------------------------------|:---------------------------------------------------|:-------------------|
| `version`                       | Version of the `buf` CLI to use. | Latest [version][buf-releases] |
| `username`                      | Username for [logging into the BSR](https://buf.build/docs/bsr/authentication). | |
| `token`                         | API token for [logging into the BSR](https://buf.build/docs/bsr/authentication). | |
| `domain`                        | Domain for logging into the BSR, enterprise only.| `buf.build` |
| `input`                         | [Input](https://buf.build/docs/reference/inputs) for the `buf` command. | |
| `paths`                         | Limit to specific files or directories (separated by newlines). | |

> | `exclude_imports`               | Exclude imports. | |

Reviewer 2: Exclude imports from what specifically? Needs more explanation

Author: Updated.

> | `exclude_paths`                 | Exclude specific files or directories (separated by newlines). | |

Reviewer 2: Exclude paths from what?

Going even further, perhaps neither of these options should exist either, as I think they only apply for lint and breaking (and build?)

Author: Exclude imports is a requested feature for buf-breaking-action: https://github.com/bufbuild/buf-breaking-action/issues/14

          token: ${{ secrets.BUF_TOKEN }}
```

### Default behavior

When you push a Git commit, tag, or branch to GitHub, the action will [push named modules to the BSR](https://buf.build/docs/bsr/module/publish) using `buf push`.

> On a pull request, the action will run all checks and post a [summary comment](#summary-comment) on the PR using `buf build`, `buf lint`, `buf format`, and `buf breaking`.

Reviewer: sorry, I messed this up the first time

```suggestion
On a pull request, the action will run all checks (using `buf build`, `buf lint`, `buf format`, `buf breaking`) and then post a [summary comment](#summary-comment) on the PR.
```

When you delete a Git branch or tag, the action will archive the corresponding label on the BSR.

> This behavior can be configured to fit your desired workflow.

Reviewer: I think we can remove this sentence since the next section is configuration.

```suggestion
```

### Configuration

> To customize the behavior of the action, you can set the following inputs in the workflow file.
> Add these inputs under the `with` section of the `uses` step in the workflow file.

Reviewer: If we are going to call it "parameter" in the table (and I think we should because otherwise it is confusing with the parameter we have that is called "input") then we need to call it "parameter" everywhere in this readme. This is just one occurrence, please check the full readme for necessary renames.

```suggestion
To customize the behavior of the action, you can set the following parameters in the workflow file.
Add these parameters under the `with` section of the `uses` step in the workflow file.
```

Author: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs
They use the term input parameters, so parameters is correct here.

| `lint`                          | Whether to run the linting step. | Runs on pushes to Git PR |
| `paths`                         | Limit to specific files or directories (separated by newlines). | |
| `pr_comment`                    | Comment the results on the pull request. | Only on pull requests |
| `push`                          | Whether to run the push step. | Runs on Git pushes |
| `push_disable_create`           | Disables repository creation if it does not exist. | False |
| `setup_only`                    | Setup only the `buf` environment, optionally logging into the BSR, but without executing other commands. | |
| `token`                         | API token for [logging into the BSR](https://buf.build/docs/bsr/authentication). | |
| `username`                      | Username for [logging into the BSR](https://buf.build/docs/bsr/authentication). | |
| `version`                       | Version of the `buf` CLI to use. | Latest [version][buf-releases] |

### Skip the breaking change detection step

By default, the action runs the breaking change detection step on every pull request.
To skip this step, add the label `buf skip breaking` to the PR.

> ![Skip breaking changes example](./static/img/skip-breaking-example.png "Skip breaking changes example")

Reviewer: This screenshot doesn't really demonstrate the breaking check being skipped to me. I know we don't get a nice GitHub action skipped status we can show since we are running these all in one check, but wondering if we can show a screenshot that leverages our summary comment to show this. here is my proposal:

<img width="500" alt="Screenshot 2024-06-28 at 4 29 03 PM" src="https://github.com/bufbuild/buf-action/assets/754768/f3168a68-ae49-4d0f-b63e-45733a4cb082">

Author: The proportions on the image will render very weird as its always the full width. Need to prefer landscape over portrait images.
<img width="838" alt="Screenshot 2024-06-28 at 4 53 11 PM" src="https://github.com/bufbuild/buf-action/assets/3036610/1f54923b-e81f-4319-871a-56106b191a82">

Author: What about updating the PR comment to show the breaking skipped. And maybe include the previous commit with a green check?

Author: I'll have a play with it.

Author: Updated with an example. Annoyingly we don't show the breaking skipped anymore, it gets dropped from the status. Will have a think how to fix that as would be nice to show the step did run before.

Ensure the workflow file includes the `pull_request` event types `labeled` and `unlabeled` so checks re-run on label changes.

> To disable this behavior, override the action input `breaking`.
> 
> See [examples/disable-skip/buf-ci.yaml](examples/disable-skip/buf-ci.yaml) for an example.

Reviewer: ```suggestion

To disable the ability to skip breaking change checks via a label, set the `breaking` parameter to the value `${{ github.event_name == 'pull_request' }}` so it runs on all PRs.
See [examples/disable-skip/buf-ci.yaml](examples/disable-skip/buf-ci.yaml) for an example.
```

          token: ${{ secrets.BUF_TOKEN }}
```

### Inputs

> | Parameter                       | Description                                        | Default            |

Author: It might be useful to have the type? We could also possible break these input down into sections.

By default, the action runs the breaking change detection step on every pull request.
To skip this step, add the label `buf skip breaking` to the PR.

> ![Skip breaking changes example](./static/img/skip-breaking-example2.png "Skip breaking changes example")

Reviewer: new image is good, can you remove the `2` from the name and delete the old image?

> +         comment: false

Reviewer 2: This is wrong now I believe, its pr_comment

> ### Skip checks on commit messages

Reviewer 2: I didn't know what this meant when I first read it, I was like "skip if...there's a commit with only a message? what does that mean?" I think this might be better phrased as "Skip checks if commit message matches a specific pattern" or the like

See the [push-on-changes.yaml](examples/push-on-changes/buf-ci.yaml) example.

### Example workflows

> 

Author: Did the migration step need to be removed? https://github.com/bufbuild/buf-action/pull/37/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R279

Reviewer: I don't remember any discussion of removing that.
